core/domain/email_manager_test.py

```diff
@@ -6635,3 +6637,68 @@ def test_not_mergeable_change_list_email_is_sent_correctly(self) -> None:
                 'Thanks!',
                 messages[0].html
             )
+
+
+class MailchimpSecretTest(test_utils.GenericTestBase):
+    """Tests for the verify_mailchimp_secret."""
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.swap_webhook_feconf_return_secret = self.swap(
+            feconf, 'MAILCHIMP_WEBHOOK_SECRET', 'secret')
+        self.swap_webhook_feconf_return_none = self.swap(
+            feconf, 'MAILCHIMP_WEBHOOK_SECRET', None)
+        self.swap_webhook_secrets_return_none = self.swap_to_always_return(
+            secrets_services, 'get_secret', None)
+        self.swap_webhook_secrets_return_secret = self.swap_with_checks(
+            secrets_services,
+            'get_secret',
+            lambda _: 'secret',
+            expected_args=[
+                ('MAILCHIMP_WEBHOOK_SECRET',),
+                ('MAILCHIMP_WEBHOOK_SECRET',),
+            ]
+        )
+
+    def test_cloud_secrets_return_none_feconf_return_secret_passes(
+        self
+    ) -> None:
+        with self.swap_webhook_feconf_return_secret:
+            with self.swap_webhook_secrets_return_none:
+                with self.capture_logging(min_level=logging.WARNING) as logs:
+                    self.assertTrue(
+                        email_manager.verify_mailchimp_secret('secret'))
+                    self.assertFalse(
+                        email_manager.verify_mailchimp_secret('not-secret'))
+                    self.assertEqual(
+                        [
+                            'Cloud Secret Manager is not working.'
```

Comment:
Why are we printing "not working" when it's working? Ditto for the last test in this file.

Also I feel like the test names could be clarified -- you aren't really just testing that it's passing, you're testing other stuff here like the logs, right?

Reply:
Done, I've modified it a bit.